docs/proposals/override-values-inside-configurations-of-different-data-formats/README.md

```diff
@@ -0,0 +1,415 @@
+   Finally, we get a new config.json with JSON operation.
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: example-configmap
+  namespace: default
+data:
+  config.json: |
+    {
+      "keyA": "valueA",
+      "keyC": "newly added value",
+      "keyE": "valueE",
+      "keyF": "valueD",            
+      "keyG": "valueE",             
+      "keyH": "valueH",
+      "keyI": "valueI",
+      "keyJ": "valueJ",
+      "keyK": "valueK"
+    }
+```
+### YAML Implementation
+#### Plan 1: directly convert to and back from JSON
+If YAML directly converts to JSON and convert back with ("sigs.k8s.io/yaml"), it will lose some data type information.
+1. Dates and Times
+   ```
+   dob: 1979-05-27T07:32:00Z
+   date_of_birth: 1979-05-27
+   
+   dob: time.Time
+   date_of_birth: time.Time
+   
+   # after transformation
+   
+   dob: string
+   date_of_birth: string
+   
+   ```
+2. Supports anchors (&) and aliases (*) to reference and reuse values
+
+   ```
+   default: &default
+     name: Alice
+     age: 30
+   employee1:
+     <<: *default
+     role: Developer
+   employee2:
+     <<: *default
+     role: Designer
+   
+   # after transformation
+   
+   default:
+     age: 30
+     name: Alice
+   employee1:
+     age: 30
+     name: Alice
+     role: Developer
+   employee2:
+     age: 30
+     name: Alice
+     role: Designer
+     
+   ```
+3. Explicit Type Tags
+
+   ```
+   number: !!int 12345
+   string: !!str 6789
+   
+   
+   # after transformation
+   
+   number: 12345
+   string: 6789
+   ```
+### Can be applied if: 
+
+(1) Do not consider the situation described above 
+
+(2) Write cases to deal with different types **(large maintenance costs)**
+
+#### Plan 2: ytt
+
+Use the third party libraries **ytt**, supporting the overlays https://carvel.dev/ytt/docs/v0.50.x/ytt-overlays/ to implement and json operation similar operations. Implementation has its own specific syntax for ytt. 
+
+Documents: https://carvel.dev/ytt/docs/v0.50.x/ytt-overlays/ 
+
+Joining methods: https://github.com/carvel-dev/ytt/blob/develop/examples/integrating-with-ytt/apis.md#as-a-go-module
+
+### Test Plan
+#### UT
+- Add unit tests to cover the new functions.
+#### E2E
++ Write proposal in `coverage_docs/overridepolicy_test.md` : deployment PlaintextObjectOverrider testing;
++ Use ginkgo to complete the code `overridepolicy_test.go`.
+## Alternatives
+Given that other most of data formats can be converted to JSON. So, they can be converted to JSON first, do JSON operations, and converts back.
+There are two API designs to achieve this:
+### (1) Each data format has different name with same struct
+```go
+type Overriders struct {
+	...
+  // JSONPlaintextOverrider represents the rules dedicated to handling json object overrides
+	// +optional
+	JSONPlaintextOverrider []PlaintextObjectOverrider `json:"jsonPlaintextOverrider,omitempty"`
+	// YAMLPlaintextOverrider represents the rules dedicated to handling yaml object overrides
```

Comment:
<img width="574" alt="image" src="https://github.com/user-attachments/assets/8ad47c95-fde0-44d4-aa7f-835a460825b4">


The format here seems to have some issues.